Cargo.lock

```diff
@@ -1,3768 +0,0 @@
-# This file is automatically @generated by Cargo.
```

Comment:
Was this a conscious decision to remove the lock file? I am uncertain if I approve:

* Local builds are getting less reproducible
* We can no longer check in the lock file, which version works.

I know they say that libraries should not have a lock-file, but we have a hybrid case, we want to be able to build the example binaries.

Reply:
I added it again for now, we could discuss that another time.